src/server/game/Entities/Player/Player.cpp

```diff
@@ -19538,6 +19538,36 @@ void Player::AddInstanceEnterTime(uint32 instanceId, time_t enterTime)
         _instanceResetTimes.insert(InstanceTimeMap::value_type(instanceId, enterTime + HOUR));
 }
 
+WorldSafeLocsEntry const* Player::GetInstanceEntrance(uint32 targetMapId)
+{
+    if (GetMapId() == targetMapId)
+        return nullptr;
```

Comment:
Fetching the instance entrance of a player that is already inside the instance doesn't seem entirely unreasonable, feel like this scenario should be supported.